src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedAutoPropAccessorSymbol.cs

```diff
@@ -0,0 +1,205 @@
+
+        public override Accessibility DeclaredAccessibility => _property.DeclaredAccessibility;
+
+        public override bool IsStatic => _property.IsStatic;
+
+        public override bool IsVirtual => _property.IsVirtual;
+
+        public override bool IsOverride => _property.IsOverride;
+
+        public override bool IsAbstract => _property.IsAbstract;
+
+        public override bool IsSealed => _property.IsSealed;
+
+        public override bool IsExtern => _property.IsExtern;
+
+        public override ImmutableHashSet<string> ReturnNotNullIfParameterNotNull => ImmutableHashSet<string>.Empty;
+
+        internal override bool HasSpecialName => _property.HasSpecialName;
+
+        internal override MethodImplAttributes ImplementationAttributes => MethodImplAttributes.Managed;
+
+        internal override bool HasDeclarativeSecurity => false;
+
+        internal override MarshalPseudoCustomAttributeData? ReturnValueMarshallingInformation => null;
+
+        internal override bool RequiresSecurityObject => false;
+
+        internal override CallingConvention CallingConvention
+            => IsStatic ? CallingConvention.Default : CallingConvention.HasThis;
+
+        internal override bool GenerateDebugInfo => false;
+
+        public override DllImportData? GetDllImportData() => null;
+
+        internal override ImmutableArray<string> GetAppliedConditionalSymbols()
+            => ImmutableArray<string>.Empty;
+
+        internal override IEnumerable<SecurityAttribute> GetSecurityInformation()
+            => Array.Empty<SecurityAttribute>();
+
+        internal override bool IsMetadataNewSlot(bool ignoreInterfaceImplementationChanges = false) => IsVirtual && !IsOverride;
+
+        internal override bool IsMetadataVirtual(bool ignoreInterfaceImplementationChanges = false) => IsVirtual;
+
+        internal override bool SynthesizesLoweredBoundBody => true;
+
+        internal override void GenerateMethodBody(TypeCompilationState compilationState, DiagnosticBag diagnostics)
+        {
+            Debug.Assert(!(_property.BackingField is null));
+            var F = new SyntheticBoundNodeFactory(this, this.GetNonNullSyntaxNode(), compilationState, diagnostics);
+            F.CurrentFunction = this;
+
+            if (_accessorKind == AccessorKind.Get)
+            {
+                // Method body:
+                //
+                // {
+                //      return this.<>backingField;
+                // }
+                F.CloseMethod(F.Block(F.Return(F.Field(F.This(), _property.BackingField))));
+            }
+            else
+            {
+                // Method body:
+                //
+                // {
+                //      this.<>backingField = value;
+                // }
+                F.CloseMethod(F.Block(
+                    F.Assignment(F.Field(F.This(), _property.BackingField), F.Parameter(Parameters[0])),
+                    F.Return()));
+            }
+        }
+    }
```

Comment:
>} [](start = 4, length = 1)

I would like to suggest that we either test all scenarios supported generalizable approach, or we limit implementation to what we need to support and can test today.

Reply:
If you decide to "shrink" implementation, feel free to add asserts around what we support in the constructor.

---
In reply to: [443079008](https://github.com/dotnet/roslyn/pull/44883#discussion_r443079008) [](ancestors = 443079008)